[M6 Orders With Coupons] Expandable product cards in order creation - part 3

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->
## Expandable product cards in order creation - part 3
This is part 3 of the following PRs:
* Part 1: https://github.com/woocommerce/woocommerce-android/pull/9851 
* Part 2: https://github.com/woocommerce/woocommerce-android/pull/9852
* **Part 3: https://github.com/woocommerce/woocommerce-android/pull/10018**

#### ⚠️ Do not merge

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds the remaining updates to the expandable product cards, as per design specs: FTOWxedKUyDbBdOrxRfMyI-fi-14%3A9617 
* Item amount picker
* Disabling buttons during order update
* Redirecting to discount edit screen
* Showing total prices with currency symbol

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
1. Create a new order and add some items. 
2. Verify expandable product cards match the designs.
3. Verify the amount picker updates the order as expected.
4. Verify "+ Add discount" and "Edit discount" buttons work as expected
5. Verify "Remove product" button works as expected
6. Verify that after applying a discount the totals are displayed correctly on the product card.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/woocommerce/woocommerce-android/assets/4527432/0577f84e-4b80-4d60-8035-1607514b62af

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Update collapsed state UI

samiuelson committed:
Update collapsed state's quantity and total row

samiuelson committed:
Improve item subtotal and total formatting

samiuelson committed:
Fix detekt issues

samiuelson committed:
Add navigation to discount screen

samiuelson committed:
Merge branch 'hack/expandable-product-cards-ui' into hack/expandable-product-cards-ui-2

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt

samiuelson committed:
Implement item amount picker (stepper view)

samiuelson committed:
Fix detekt issue

samiuelson committed:
Clean up code

samiuelson committed:
Merge branch 'hack/expandable-product-cards-ui' into hack/expandable-product-cards-ui-2

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt

samiuelson committed:
Merge branch 'hack/expandable-product-cards-ui' into hack/expandable-product-cards-ui-2

samiuelson committed:
Fix collapsed state total and discount amount showing

AnirudhBhat commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt:
```diff
@@ -1190,6 +1190,16 @@ class OrderCreateEditViewModel @Inject constructor(
         tracker.track(AnalyticsEvent.TAX_RATE_AUTO_TAX_RATE_CLEAR_ADDRESS_TAPPED)
     }
 
+    fun onDiscountButtonClicked(item: Order.Item) {
```
nitpick: ❓ We could write tests for this logic. WDYT?

samiuelson replied:
Good point. I'm going to add tests in the next PR.

AnirudhBhat reviewed (approved):
Thank you for your work on this, @samiuelson. The core functionality seems solid and aligns with expectations. However, I've spotted a couple of UI nuances and a minor code suggestion:

Regarding the arrow animation on the expandable product card: there seems to be an inconsistency in its direction. Intuitively, a downward-pointing arrow would imply that the card can be expanded, whereas an upward-pointing arrow would suggest the card can be collapsed. Right now, the behavior is reverse. WDYT?

I've noticed that when the product card is collapsed, the applied discount isn't visible. I've attached a video for clarity.


https://github.com/woocommerce/woocommerce-android/assets/1331230/7397c308-397e-4545-886a-98c732566d82



I've also made a slight code observation, but it's non-critical.

samiuelson committed:
Switch chevron target angles in collapsed/expanded states

samiuelson commented:
Thanks for review @AnirudhBhat !

> Regarding the arrow animation on the expandable product card: there seems to be an inconsistency in its direction. Intuitively, a downward-pointing arrow would imply that the card can be expanded, whereas an upward-pointing arrow would suggest the card can be collapsed. Right now, the behavior is reverse. WDYT?

Good idea. I switched the target angles in collapsed/expanded states in b3376bde1de36074db9292407da132733511eb99.

> I've noticed that when the product card is collapsed, the applied discount isn't visible. I've attached a video for clarity.

Hmm, I've added this yesterday already in f53511e7cdf8b89b0ffed89e9e7a9eb0c5456002. Is there a chance you were testing an outdated branch?

samiuelson committed:
Clean up code
